product/gql_mutations.py

```diff
@@ -61,6 +62,14 @@ def create_or_update_product(user, data):
     deductibles = extract_deductibles(data)
     ceilings = extract_ceilings(data)
 
+    incoming_code = data.get('code')
+    current_product = Product.objects.filter(uuid=product_uuid)
```

Comment:
You are missing .first() here

```suggestion
    current_product = Product.objects.filter(uuid=product_uuid).first()
```